### PR TITLE
Create mergesort.c

### PR DESCRIPTION
**Pull request by soubh1k**
mergesort.c added

### CODE REVIEW

File: cpp/quicksort.c [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+void printArray(int arr[], int size)
+{
+    int i;
+    for (i=0; i < size; i++)
+        printf("%d ", arr[i]);
+    printf("\n");
+}
+ 
+// Driver program to test above functions
+int main()
```
Comment (thecodershub) [inline cpp/quicksort.c]: Add a function "test/check" to actually sort an unsorted, sorted, empty and 1 element array. Assert results there. Call that in main.

Comment (soubh1k) [inline cpp/quicksort.c]: Ok! Done.
